workshops/test/test_util.py

```diff
@@ -321,3 +328,104 @@ def test_upload_existing_user_existing_task(self):
         self.assertEqual(tasks_pre, tasks_post)
         self.assertEqual(users_pre, users_post)
         self.assertEqual(rv.status_code, 200)
+
+
+class TestEventURLNormalization(TestCase):
+    def setUp(self):
+        self.test_cases = [
+            'http://username.github.io/2015-07-13-City/',
+            'http://username.github.io/2015-07-13-City',
+            'http://username.github.io/2015-07-13-City/index.html',
+            'https://github.com/username/2015-07-13-City/',
+            'https://github.com/username/2015-07-13-City',
+            ('https://github.com/username/2015-07-13-City/blob/'
+             'gh-pages/index.html'),
+        ]
+
+        self.output = ('https://raw.githubusercontent.com/username/'
+                       '2015-07-13-City/gh-pages/index.html')
+
+    def test_normalization(self):
+        for url in self.test_cases:
+            assert normalize_event_index_url(url)[0] == self.output
+
+
+class TestParsingEventHeaders(TestCase):
+
+    # Response.status_code apparently doesn't exist by default
+    @patch.object(requests.models.Response, 'status_code', 404, create=True)
+    def test_wrong_url(self):
+        with self.assertRaises(requests.exceptions.HTTPError):
+            url = 'http://test.github.io/2015-07-13-test/'
+            parse_tags_from_event_index(url)
+
+    @patch.object(requests, 'get')
+    def test_parsing_event_index(self, mock_get):
+        mock_get.return_value.status_code = 200
+        mock_get.return_value.text = """---
+layout: workshop
+root: .
+venue: Euphoric State University
+address: Highway to Heaven 42, Academipolis
+country: USA
+language: us
+latlng: 36.998977, -109.045173
+humandate: Jul 13-14, 2015
+humantime: 9:00 - 17:00
+startdate: 2015-07-13
+enddate: 2015-07-14
+instructor: ["Hermione Granger", "Harry Potter", "Ron Weasley",]
+helper: ["Peter Parker", "Tony Stark", "Natasha Romanova",]
```

Comment:
These are legitimate people in our tests. Greg came up with this, and I think it's brilliant :-)